doc/how-to-define-samplers.md

```diff
@@ -0,0 +1,72 @@
+# How to test functions with user-defined types: definition of a sampler
```

Comment:
Some parts of this file seem to have a wider scope than learn-ocaml autogen. Let us avoid redundancy and point to the corresponding documentation of learn-ocaml when it is possible.